functions/src/modules/SQR.functions.ts

```diff
@@ -303,118 +299,107 @@ export const processSubmission = functions.database
 /////////////////////////////////////////////////
 export const importSpreadSheetData = functions.https.onRequest(
   async (req, res) => {
-    const auth = await google.auth.getClient({
-      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
-    });
-
-    const spreadsheetId = functions.config().sqr.spreadsheetId;
-    const spreadsheet = new GoogleSpreadsheet(spreadsheetId);
-
-    const token = await auth.getAccessToken();
-    spreadsheet.setAuthToken(token);
-
-    const getInfo = promisify(spreadsheet.getInfo);
-    const data = await getInfo();
-
-    let AllotmentsSheet, SubmissionsSheet;
-
-    data.worksheets.forEach(worksheet => {
-      if (worksheet.title === 'Submissions') SubmissionsSheet = worksheet;
-      if (worksheet.title === 'Allotments') AllotmentsSheet = worksheet;
-    });
-
-    const getSubmissions = promisify(SubmissionsSheet.getRows);
-
-    // submissions = Submissions sheet rows
-    const submissions = await getSubmissions();
-    submissions.forEach(row => {
-      const serial = row['submissionserial'];
+    const spreadsheetId = functions.config().sqr.spreadsheet_id;
+
+    // Regex for validating file names in both sheets
+    const audioFileRegex = /(\w+)-\d+.*/;
+
+    ////////////////////////
+    //     Submissions
+    ////////////////////////
+    const submissionsSheet = new GoogleSheets(spreadsheetId, ISoundQualityReportSheet.Submissions);
+    const submissionsRows = await submissionsSheet.__getRows();
+    for (const row of submissionsRows) {
+      const audioFileName = row['Audio File Name'];
+      const match = audioFileRegex.exec(audioFileName);
+      if (!match) continue;
+
+      const list = match[1];
+      // const serial = row['Submission Serial'];
+
+      const fileRef = await db.ref(`/files/${list}/${audioFileName}/soundQualityReporting`).once('value');
+      const audioFile = fileRef.val();
+      let token;
+      if (fileRef.exists()) {
+        token = audioFile.token;
+      } else {
+        console.warn(`Token not found!`)
```

Comment:
This check and message do not correlate. All this block above is not clear, please rewrite it and make comments on WHY are we doing this.

Reply:
Token should be extracted from the Update Link column and overwritten in the database.